### ISSUE TITLE
Clean architecture: Make

### ISSUE THREAD

**Issue by JohannesLichtenberger** [sirixdb/sirix#519]

**Comment by pernelkanic** [comment]
can i work on this issue

**Comment by JohannesLichtenberger** [comment]
@pernelkanic of course :) the enum in question is `PageKind` plus interface(s).

**Comment by JohannesLichtenberger** [comment]
Yes :-) maybe it would be even better to wrap the configuration in a context (simple data wrapper), but I guess it's ok :-)

**Comment by JohannesLichtenberger** [comment]
@pernelkanic can you open a PR?

**Comment by pernelkanic** [comment]
I have opened the PR, requesting review to know if there should be any changes.

**Comment by JohannesLichtenberger** [comment]
Currently it fails to compile, because I think an export missing.

**Comment by pernelkanic** [comment]
what should be done?

**Comment by pernelkanic** [comment]
It seems like org.jcp.xml.dsig.internal.dom.Utils is import that is causing the issue , it is considered internal and used in java.xml.crypto

**Comment by pernelkanic** [comment]
Does removing it fix the issue?

**Comment by JohannesLichtenberger** [comment]
Why do you need java.xml.crypto for the changes? Can you remove the import and update your PR?

**Comment by pernelkanic** [comment]
i have updated the pr
